app/feedstock/[id]/dataset-repr.tsx

```diff
@@ -0,0 +1,76 @@
+import { fetcher } from '@/utils/fetcher'
+import { Info } from '@carbonplan/icons'
+import React from 'react'
+import useSWR from 'swr'
+import { Alert, Box, Card, Flex, Text } from 'theme-ui'
+
+interface DatasetReprProps {
+  url: string
+}
+
+interface DatasetRepr {
+  html: string
+  url: string
+  sanitized_url: string
+}
+
+export const DatasetRepr: React.FC<DatasetReprProps> = ({ url }) => {
+  const { data, error, isLoading } = useSWR<DatasetRepr>(
+    `/api/dataset-repr?url=${url}`,
+    fetcher,
+    {
+      suspense: true,
+      fallbackData: { html: '', url: '', sanitized_url: '' },
+      dedupingInterval: 3600,
+    },
+  )
+
+  if (error) {
+    return (
+      <Alert variant='primary' sx={{ p: 3, borderRadius: 'default' }}>
+        <Flex sx={{ flexDirection: 'row', gap: 4 }}>
+          <Info />
+
+          <Text>Failed to load dataset representation: {error.message}</Text>
+        </Flex>
+      </Alert>
+    )
+  }
+
+  if (isLoading) {
+    return (
+      <>
+        <Card sx={{ p: 3, borderRadius: 'default', width: '100%' }}>
+          <Box
+            sx={{
+              height: '200px',
+              bg: 'muted',
+              width: '100%',
+              borderColor: 'muted',
+              borderRadius: 'default',
+            }}
+          />
+        </Card>
+      </>
+    )
+  }
+
+  return (
+    <>
+      <Card sx={{ p: 3, borderRadius: 'default', width: '100%' }}>
+        {data?.html ? (
+          <Box
+            dangerouslySetInnerHTML={{ __html: data?.html }}
```

Comment:
The output doesn't play vary nicely with darkmode -- should we just remove that option from the header?
![CleanShot 2024-10-03 at 17 09 58@2x](https://github.com/user-attachments/assets/89a05dc9-57fb-4063-a907-fa499edf6228)